### ISSUE TITLE
XML library instead of string concatenation 

### ISSUE THREAD

**Issue by AzaelCicero** [carla-simulator/scenario_runner#766]
**Is your feature request related to a problem? Please describe.**
Currently srunner/scenariomanager/result_writer.py uses string concatenation to format XML with results. I propose to use XML library to this. Then for example we won't have to worry about security vulnerability i.e. https://owasp.org/www-project-top-ten/2017/A4_2017-XML_External_Entities_(XXE).

**Describe the solution you'd like**
Replace string concatenation with XML library. 

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by fabianoboril** [comment]
Hi,
would you be able to prepare a draft PR for this?

**Comment by AzaelCicero** [comment]
Yes, I would be able to prepare such PR if you think it will be viable addition to Scenario Runner.
